### ISSUE TITLE
Fresco OutofMemory error

### ISSUE THREAD

**Issue by ytrjp** [facebook/fresco#319]
java.lang.OutOfMemoryError: Failed to allocate a 3499212 byte allocation with 1632968 free bytes and 1594KB until OOM
            at dalvik.system.VMRuntime.newNonMovableArray(Native Method)
            at android.graphics.Bitmap.nativeCreate(Native Method)
            at android.graphics.Bitmap.createBitmap(Bitmap.java:939)
            at android.graphics.Bitmap.createBitmap(Bitmap.java:912)
            at android.graphics.Bitmap.createBitmap(Bitmap.java:879)
            at com.facebook.imagepipeline.memory.BitmapPool.alloc(BitmapPool.java:55)
            at com.facebook.imagepipeline.memory.BitmapPool.alloc(BitmapPool.java:30)
            at com.facebook.imagepipeline.memory.BasePool.get(BasePool.java:260)
            at com.facebook.imagepipeline.bitmaps.ArtBitmapFactory.doDecodeStaticImage(ArtBitmapFactory.java:131)
            at com.facebook.imagepipeline.bitmaps.ArtBitmapFactory.decodeJPEGFromPooledByteBuffer(ArtBitmapFactory.java:119)
            at com.facebook.imagepipeline.bitmaps.PlatformBitmapFactory.decodeJPEGFromPooledByteBuffer(PlatformBitmapFactory.java:106)
            at com.facebook.imagepipeline.decoder.ImageDecoder.decodeJpeg(ImageDecoder.java:135)
            at com.facebook.imagepipeline.decoder.ImageDecoder.decodeImage(ImageDecoder.java:83)
            at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.doDecode(DecodeProducer.java:166)
            at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.access$000(DecodeProducer.java:88)
            at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder$1.run(DecodeProducer.java:112)
            at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder$1.run(DecodeProducer.java:109)
            at com.facebook.imagepipeline.producers.JobScheduler.doJob(JobScheduler.java:214)
            at com.facebook.imagepipeline.producers.JobScheduler.access$000(JobScheduler.java:30)
            at com.facebook.imagepipeline.producers.JobScheduler$1.run(JobScheduler.java:68)
            at com.facebook.common.executors.SerialDelegatingExecutor.executeSingleCommand(SerialDelegatingExecutor.java:76)
            at com.facebook.common.executors.SerialDelegatingExecutor.access$000(SerialDelegatingExecutor.java:24)
            at com.facebook.common.executors.SerialDelegatingExecutor$1.run(SerialDelegatingExecutor.java:47)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:818)

**Comment by michalgr** [comment]
Can you try configuring size of the memory cache to be smaller ?

**Comment by tyronen** [comment]
Stepping back - this is a 34 MB decode you attempted. What is the URI of the original image? It seems much too large for mobile.

**Comment by Nashadelic** [comment]
@tyronen it's a 3.4 MB image: 3,499,212 bytes

**Comment by mrlongtt** [comment]
I got same issue. Here stack traces in Fabric:

```
Fatal Exception: java.lang.OutOfMemoryError
Failed to allocate a 8294412 byte allocation with 592938 free bytes and 579KB until OOM
 raw
dalvik.system.VMRuntime.newNonMovableArray (VMRuntime.java)
android.graphics.Bitmap.createBitmap (Bitmap.java:879)
com.facebook.imagepipeline.memory.BitmapPool.alloc (BitmapPool.java:55)
com.facebook.imagepipeline.memory.BitmapPool.alloc (BitmapPool.java:30)
com.facebook.imagepipeline.memory.BasePool.get (BasePool.java:264)
com.facebook.imagepipeline.bitmaps.ArtBitmapFactory.doDecodeStaticImage (ArtBitmapFactory.java:131)
com.facebook.imagepipeline.bitmaps.ArtBitmapFactory.decodeFromPooledByteBuffer (ArtBitmapFactory.java:84)
com.facebook.imagepipeline.bitmaps.PlatformBitmapFactory.decodeFromPooledByteBuffer (PlatformBitmapFactory.java:87)
com.facebook.imagepipeline.decoder.ImageDecoder.decodeStaticImage (ImageDecoder.java:114)
com.facebook.imagepipeline.decoder.ImageDecoder.decodeImage (ImageDecoder.java:92)
com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.doDecode (DecodeProducer.java:231)
com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.access$200 (DecodeProducer.java:92)
com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder$3.run (DecodeProducer.java:202)
com.facebook.common.executors.SerialDelegatingExecutor.executeSingleCommand (SerialDelegatingExecutor.java:76)
com.facebook.common.executors.SerialDelegatingExecutor.access$000 (SerialDelegatingExecutor.java:24)
com.facebook.common.executors.SerialDelegatingExecutor$1.run (SerialDelegatingExecutor.java:47)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1112)
java.lang.Thread.run (Thread.java:818)
```

**Comment by plamenko** [comment]
In both stack traces allocation fails simply because there is no memory left.
Requested 3.5 MB, available 1.6 MB.
Requested 8 MB, available 0.5 MB.

There can be many things causing the memory issues:
* You may have a leak in the application. Check the heap dump with MAT.
* You may have been displaying too many images at once, and not all of them can fit in the memory at once.
* You may need to downscale / resize the image before displaying them.
